tests/end_to_end_test/snapshot_fail_test.go

```diff
@@ -0,0 +1,141 @@
+package endtoend_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+
+	"github.com/kopia/kopia/tests/testenv"
+)
+
+func TestSnapshotFail(t *testing.T) {
+	t.Parallel()
+
+	e := testenv.NewCLITest(t)
+	defer e.Cleanup(t)
+	defer e.RunAndExpectSuccess(t, "repo", "disconnect")
+
+	e.RunAndExpectSuccess(t, "repo", "create", "filesystem", "--path", e.RepoDir)
+	e.RunAndExpectSuccess(t, "policy", "set", "--global", "--keep-latest", strconv.Itoa(1<<31-1))
+
+	parentDirOfSource := makeScratchDir(t)
+	source := filepath.Join(parentDirOfSource, "source")
+
+	// Test snapshot of nonexistent directory fails
+	e.RunAndExpectFailure(t, "snapshot", "create", source)
+
+	testenv.MustCreateDirectoryTree(t, source, testenv.DirectoryTreeOptions{
+		Depth:                  2,
+		MaxSubdirsPerDirectory: 1,
+		MaxFilesPerDirectory:   1,
+	})
```

Comment:
thought: After generating the test directory, you can always "inject" a file and a directory under `source` with well-known names, so they can be used for the tests and do not need to be searched for.

Reply:
mm in fact my old helper func used constant human readable names to help with debug. Lines of code we save in not searching will be counterbalanced by new lines to create the specific file structure though :)